rainier-core/src/main/scala/com/stripe/rainier/core/Support.scala

```diff
@@ -8,9 +8,41 @@ import com.stripe.rainier.compute._
   * and its log-jacobian.
   */
 private[rainier] trait Support {
+  import Support._
+
   def transform(v: Variable): Real
 
   def logJacobian(v: Variable): Real
+
+  def union(that: Support): Support = (this, that) match {
+    case (UnboundedSupport, _)                            => UnboundedSupport
+    case (_, UnboundedSupport)                            => UnboundedSupport
+    case (BoundedBelowSupport(_), BoundedAboveSupport(_)) => UnboundedSupport
+    case (BoundedAboveSupport(_), BoundedBelowSupport(_)) => UnboundedSupport
+    case (BoundedBelowSupport(thisMin), BoundedSupport(thatMin, thatMax)) =>
+      BoundedBelowSupport(realMin(thisMin, thatMin))
+    case (BoundedSupport(thisMin, thisMax), BoundedBelowSupport(thatMin)) =>
+      BoundedBelowSupport(realMin(thisMin, thatMin))
+    case (BoundedAboveSupport(thisMax), BoundedSupport(thatMin, thatMax)) =>
+      BoundedAboveSupport(realMax(thisMax, thatMax))
+    case (BoundedSupport(thisMin, thisMax), BoundedAboveSupport(thatMax)) =>
+      BoundedAboveSupport(realMax(thisMax, thatMax))
+    case (BoundedSupport(thisMin, thisMax), BoundedSupport(thatMin, thatMax)) =>
+      BoundedSupport(realMin(thisMin, thatMin), realMax(thisMax, thatMax))
+  }
+
+  def isDefinedAt(real: Real): Real
+}
+
+object Support {
+  private def realMin(a: Real, b: Real): Real = ((a - b).abs - (b + a)) / 2.0
```

Comment:
Neat. Should we move this into `Real` itself?

Reply:
Happy to!